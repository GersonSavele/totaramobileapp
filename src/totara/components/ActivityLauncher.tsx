/**
 * This file is part of Totara Mobile
 *
 * Copyright (C) 2019 onwards Totara Learning Solutions LTD
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * @author Jun Yamog <jun.yamog@totaralearning.com
 */

import React from "react";
import {StyleSheet, Text, TouchableOpacity, View} from "react-native";

import {Activity} from "@totara/types";
import ContentIcon from "@totara/components/ContentIcon";
import ProgressCircle from "@totara/components/ProgressCircle";
import {translate} from "@totara/locale";

/**
 * Render a preview of the activity and execute a function when component is touched
 *
 * @param item - Activity to be rendered
 * @param onPress - function to fire when component is touched
 */
type Props = {
  item: Activity,
  onPress: (activity: Activity) => void
}

const ActivityLauncher = ({item, onPress = () => {}}: Props) => {
  return(
    <TouchableOpacity style={styles.activityContainer} onPress={() => onPress(item)}>
      <ContentIcon icon={item.type} iconSize={16} size={40}/>
      <View style={styles.container}>
        <Text style={styles.topText}>{translate("totara-component.section_title_continue_learn")}</Text>
        <Text numberOfLines={1} style={styles.bottomText}>{item.itemName}</Text>
      </View>
      <View style={styles.divider}/>
      { (item.progressPercentage) &&
          <ProgressCircle size={32} progress={item.progressPercentage}/>
      }
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  activityContainer: {
    flexDirection: "row",
    backgroundColor: "#FFFFFF",
    alignItems: "center",
    paddingRight: 8,
    paddingLeft: 8,
    paddingTop: 8,
    paddingBottom: 8
  },
  container: {
    flex: 1,
    paddingLeft: 8,
    paddingTop: 4
  },
  topText: {
    fontSize: 10,
    lineHeight: 12,
    color: "#64717D"
  },
  bottomText: {
    flex: 1,
    fontSize: 12,
    paddingRight: 64,
    color: "#3D444B",
    fontWeight: "bold"
  },
  divider: {
    marginRight: 16,
    backgroundColor: "#EEEEEE",
    width: 2,
    height: 30
  }
});

export default ActivityLauncher;
