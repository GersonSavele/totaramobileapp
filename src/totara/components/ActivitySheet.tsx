/**
 * This file is part of Totara Mobile
 *
 * Copyright (C) 2019 onwards Totara Learning Solutions LTD
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * @author Jun Yamog <jun.yamog@totaralearning.com
 */

import React from "react";
import SlidingUpPanel from "rn-sliding-up-panel";
import {Image, StyleSheet, Text, View, StatusBar} from "react-native";
import {Button} from "native-base";
import {FontAwesomeIcon} from "@fortawesome/react-native-fontawesome";
import {widthPercentageToDP as wp} from "react-native-responsive-screen";

import {config} from "@totara/lib";
import {Activity} from "@totara/types";


type contextData = {
  toggleActivity: () => void,
  activitySheetVisible: boolean,
  currentActivity: Activity,
  setCurrentActivity: (activity: Activity) => void
}


const ActivitySheetContext = React.createContext<contextData>({
  toggleActivity: () => {},
  activitySheetVisible: false,
  currentActivity: { // placeholder
    id: 0,
    type: "",
    itemName: "",
  },
  setCurrentActivity: activity => {}
});

export const ActivitySheetProvider = ActivitySheetContext.Provider;
export const ActivitySheetConsumer = ActivitySheetContext.Consumer;

export const ActivitySheet = () => {

    return(
      <ActivitySheetConsumer>
        {({toggleActivity, activitySheetVisible, currentActivity}) =>
          (<SlidingUpPanel
            visible={activitySheetVisible}
            onRequestClose={toggleActivity}>
            <View style={styles.panel}>
              <View style={{paddingLeft: 10}}>
              <StatusBar hidden/>
              <Button transparent onPress={toggleActivity}>
                <FontAwesomeIcon
                  icon="times"
                  size={24}
                />
              </Button>
              </View>
              {
                (currentActivity && currentActivity.imgSrc) ?
                  <Image source={{uri: config.mobileStatic + "/public/" + currentActivity.imgSrc}}
                         style={{width: wp("100%"), height: 240}}/>
                  : null
              }
              {
                (currentActivity && currentActivity.summary) ?
                  <Text style={styles.panelContent}>
                    {currentActivity.summary}
                  </Text>
                  : null
              }
            </View>
          </SlidingUpPanel>)
        }
      </ActivitySheetConsumer>);
  };


const styles = StyleSheet.create({
  panel: {
    flex: 1,
    flexDirection: "column",
    backgroundColor: "#CECECE",
    paddingTop: 5
  },
  panelContent: {
    flex: 10,
    padding: 20,
  },
});
