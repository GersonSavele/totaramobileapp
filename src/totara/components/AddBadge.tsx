/**
 * This file is part of Totara Enterprise.
 *
 * Copyright (C) 2019 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise is provided only to Totara Learning Solutions
 * LTDâ€™s customers and partners, pursuant to the terms and
 * conditions of a separate agreement with Totara Learning
 * Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [sales@totaralearning.com] for more information.
 */

import React, { useContext } from "react";
import { View, StyleSheet } from "react-native";
// @ts-ignore no types published yet for fortawesome react-native, they do have it react so check in future and remove this ignore
import { FontAwesomeIcon } from "@fortawesome/react-native-fontawesome";

import { Status } from "@totara/types";
import ProgressCircle from "./ProgressCircle";
import { ThemeContext } from "@totara/theme";
import { iconSizes } from "@totara/theme/constants";

type Props = {
  size: number;
  offsetSize: number;
  status?: Status | number;
};

type ProgressProps = {
  size?: number;
  progress: number;
};
type IconBadgeProps = {
  size: number;
  color: string;
};

const AddBadge = ({
  status,
  children,
  size = iconSizes.sizeS,
  offsetSize = 8,
  ...otherProps
}: {
  status: Status | number;
  children?: JSX.Element;
  size: number;
  offsetSize?: number;
}) => {
  return (
    <View>
      {children}
      <RightBadge status={status} size={size} offsetSize={offsetSize} {...otherProps} />
    </View>
  );
};

const RightBadge = ({ status, size, offsetSize }: Props) => {
  const theme = useContext(ThemeContext);
  const getContainerStyle = (size: number, offsetSize: number, backgroundColor: string, borderColor: string) => {
    return StyleSheet.create({
      container: {
        top: -1 * offsetSize,
        right: -1 * offsetSize,
        position: "absolute",
        backgroundColor: backgroundColor,
        borderRadius: size * 2,
        borderWidth: size / 8,
        borderColor: borderColor,
        width: size * 2,
        height: size * 2,
        justifyContent: "center",
        alignItems: "center"
      }
    });
  };
  switch (status) {
    case Status.done: {
      const viewStyle = getContainerStyle(size, offsetSize, theme.colorSuccess, theme.colorNeutral1);
      return (
        <View style={viewStyle.container}>
          <CheckBadge size={size} color={theme.textColorLight} />
        </View>
      );
    }
    case 100: {
      const viewStyle = getContainerStyle(size, offsetSize, theme.colorSuccess, theme.colorNeutral1);
      return (
        <View style={viewStyle.container}>
          <CheckBadge size={size} color={theme.textColorLight} />
        </View>
      );
    }
    case Status.hidden: {
      const viewStyle = getContainerStyle(size, offsetSize, "#999999", theme.colorNeutral1);
      return (
        <View style={viewStyle.container}>
          <LockBadge size={size} color={theme.textColorLight} />
        </View>
      );
    }
    case Status.active: {
      // drop through default
      return null;
    }
    default: {
      if (typeof status == "number") {
        const viewStyle = getContainerStyle(size, offsetSize, theme.colorNeutral1, theme.colorNeutral1);
        return (
          <View style={viewStyle.container}>
            <ProgressBadge size={size} progress={status} />
          </View>
        );
      }
      return null;
    }
  }
};

const CheckBadge = ({ size, color }: IconBadgeProps) => <FontAwesomeIcon icon={"check"} color={color} size={size} />;

const LockBadge = ({ size, color }: IconBadgeProps) => <FontAwesomeIcon icon={"lock"} color={color} size={size} />;

const ProgressBadge = ({ size, progress }: ProgressProps) => (
  <ProgressCircle size={size ? size * 2 : 32} progress={progress} />
);

export { AddBadge, CheckBadge, LockBadge, ProgressBadge };
