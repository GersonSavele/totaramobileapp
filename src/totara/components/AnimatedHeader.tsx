/**
 * This file is part of Totara Enterprise.
 *
 * Copyright (C) 2020 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise is provided only to Totara Learning Solutions
 * LTDâ€™s customers and partners, pursuant to the terms and
 * conditions of a separate agreement with Totara Learning
 * Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [sales@totaralearning.com] for more information.
 */

import { FontAwesomeIcon } from "@fortawesome/react-native-fontawesome";
import { useFocusEffect } from "@react-navigation/native";
import { translate } from "@totara/locale";
import { fontSizes, fontWeights, paddings } from "@totara/theme/constants";
import React, { useCallback, useState } from "react";
import { StyleSheet, Text, TouchableOpacity, View, StatusBar, Platform } from "react-native";
import Animated, { Extrapolate, interpolate, call, useCode } from "react-native-reanimated";
import { useSafeAreaInsets } from "react-native-safe-area-context";

const STATUSBAR_HEIGHT = 50;
const TOPNAVI_OFFSET = 250;
const ACTION_WIDTH = 40;
const HEIGHT = TOPNAVI_OFFSET + STATUSBAR_HEIGHT;

type AnimatedHeaderProps = { title: string; subTitle?: string; scrollValue?: any; leftAction: any };

const getIcon = () => Platform.OS === "ios" ? "chevron-left" : "arrow-left";

const AnimatedHeader = ({ title, subTitle, scrollValue, leftAction }: AnimatedHeaderProps) => {
  const safeArea = useSafeAreaInsets();
  const isFloating = !!scrollValue;
  const [dark, setDark] = useState(false);

  useCode(() => {
    return call([scrollValue], (values) => {
      const [value] = values;
      setDark(TOPNAVI_OFFSET < value);
    });
  }, [scrollValue]);

  const transparentToOpaqueInterpolate = interpolate(scrollValue, {
    inputRange: [TOPNAVI_OFFSET, TOPNAVI_OFFSET + STATUSBAR_HEIGHT],
    outputRange: [0, 1],
    extrapolate: Extrapolate.CLAMP
  });

  const opaqueToTransparentInterpolate = interpolate(scrollValue, {
    inputRange: [TOPNAVI_OFFSET, TOPNAVI_OFFSET + STATUSBAR_HEIGHT],
    outputRange: [1, 0],
    extrapolate: Extrapolate.CLAMP
  });

  useFocusEffect(
    useCallback(() => {
      StatusBar.setBarStyle(dark ? "dark-content" : "light-content", true);
      return () => {
        StatusBar.setBarStyle("dark-content", true);
      };
    }, [])
  );

  return (
    <>
      <Animated.View
        testID={"animated-header-container"}
        style={{
          paddingTop: safeArea.top,
          marginBottom: isFloating ? -STATUSBAR_HEIGHT - safeArea.top : 0,
          height: STATUSBAR_HEIGHT + safeArea.top,
          backgroundColor: Animated.color(255, 255, 255, transparentToOpaqueInterpolate), //RGB color, 255 being white
          opacity: 1,
          zIndex: 200,
          flexDirection: "row"
        }}>
        <TouchableOpacity
          accessibilityLabel={translate("general.accessibility_go_back")}
          accessibilityRole={"button"}
          testID={"animated-header-backbutton"}
          onPress={leftAction}
          style={styles.leftAction}>
          <Animated.View
            testID={"animated-header-backbutton-black"}
            style={[styles.backIcon, { opacity: transparentToOpaqueInterpolate }]}>
            <FontAwesomeIcon icon={getIcon()} color={"black"} />
          </Animated.View>
          <Animated.View
            testID={"animated-header-backbutton-white"}
            style={[styles.backIcon, { opacity: opaqueToTransparentInterpolate }]}>
            <FontAwesomeIcon icon={getIcon()} color={"white"} />
          </Animated.View>
        </TouchableOpacity>

        <Animated.View
          testID={"animated-header-title-container"}
          style={{ flex: 1, opacity: transparentToOpaqueInterpolate }}>
          <Text numberOfLines={1} testID={"animated-header-title"} style={styles.title}>
            {title}
          </Text>
          {subTitle && (
            <Text testID={"animated-header-subtitle"} style={styles.subTitle}>
              {subTitle}
            </Text>
          )}
        </Animated.View>
        <View style={{ width: ACTION_WIDTH }}></View>
      </Animated.View>
    </>
  );
};

const styles = StyleSheet.create({
  leftAction: {
    width: ACTION_WIDTH,
    paddingLeft: paddings.paddingL,
    alignContent: "center",
    justifyContent: "center",
    flexDirection: "row",
    alignItems: "center"
  },
  title: {
    textAlign: "center",
    fontWeight: fontWeights.fontWeightBold,
    fontSize: fontSizes.fontSizeM
  },
  subTitle: {
    textAlign: "center",
    fontSize: fontSizes.fontSizeS,
    paddingTop: paddings.paddingS
  },
  backIcon: {
    justifyContent: "center",
    alignItems: "center",
    alignSelf: "center",
    position: "absolute"
  }
});

export { AnimatedHeader, HEIGHT };
