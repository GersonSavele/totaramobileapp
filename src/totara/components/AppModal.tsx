/**
 * This file is part of Totara Mobile
 *
 * Copyright (C) 2019 onwards Totara Learning Solutions LTD
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * @author: Kamala Tennakoon <kamala.tennakoon@totaralearning.com>
 */

import React, { useContext, useState } from "react";

import { translate } from "@totara/locale";
import { AuthContext, Setup } from "@totara/auth/AuthContext";
import { InfoModal} from "./infoModal"
import PrimaryButton from "./PrimaryButton";
import { config } from "@totara/lib";

enum Compatible {
  Api = 1
}

const AppModal = () => {
  const {setup} = useContext(AuthContext);
  const isShowIncompatibleApi = !isValidApiVersion(setup);
  const [isVisible, setIsVisible] = useState(isShowIncompatibleApi);
  if(isVisible) 
    return (
    <InfoModal title={translate("message.sorry")} description={translate("message.incompatible_api")} imageType={"url_not_valid"} visible={isVisible}>
      <PrimaryButton  text={translate("general.try_again")} onPress={() => { setIsVisible(!isVisible) }} />
    </InfoModal>
    );
  else 
    return null;
};

//TODO-Need to integrate correct logic
export const isValidApiVersion = (setup?: Setup) => {
  if (setup && setup.apiVersion) {
    const compatibilityList = isCompatible(setup.apiVersion)
    return compatibilityList.length > 0
  }
  return false;
};

export const isCompatible = (version: string) => {
  if (config.minApiVersion <= version)
    return [Compatible.Api]
  else 
    return []
};

export default AppModal;