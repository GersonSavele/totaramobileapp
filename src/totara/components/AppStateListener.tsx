/**
 * This file is part of Totara Enterprise.
 *
 * Copyright (C) 2019 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise is provided only to Totara Learning Solutions
 * LTDâ€™s customers and partners, pursuant to the terms and
 * conditions of a separate agreement with Totara Learning
 * Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [sales@totaralearning.com] for more information.
 */

import React, { ReactNode } from "react";
import { AppState, AppStateStatus } from "react-native";
import { Log } from "@totara/lib";

type Props = {
  onActive?: () => void;
  onBackground?: () => void;
  onInactive?: () => void;
  children?: ReactNode;
};

class AppStateListener extends React.Component<Props> {
  static defaultProps = {
    onActive: () => Log.debug("App is in Active/Foreground Mode."),
    onBackground: () => Log.debug("App is in Background Mode."),
    onInactive: () => Log.debug("App is in Inactive mode.")
  };

  constructor(props: Props) {
    super(props);
  }
  state = {
    appState: AppState.currentState
  };

  componentDidMount() {
    AppState.addEventListener("change", this.handleAppStateChange);
  }

  componentWillUnmount() {
    AppState.removeEventListener("change", this.handleAppStateChange);
  }

  /**
   * @AppStateChange : Here, will trigger when app is background/inactive and foreground.
   * @parameter : When app goes background and foreground, next app state.
   * @return : null
   */

  handleAppStateChange = (nextAppState: AppStateStatus) => {
    this.setState({ appState: nextAppState });

    if (nextAppState === "active") {
      // Do something here on app active foreground mode.
      this.props.onActive!();
    }
    if (nextAppState === "background") {
      // Do something here on app background.
      this.props.onBackground!();
    }
    if (nextAppState === "inactive") {
      // Do something here on app inactive mode.
      this.props.onInactive!();
    }
  };

  render() {
    return this.props.children != null ? this.props.children : null;
  }
}

export default AppStateListener;
