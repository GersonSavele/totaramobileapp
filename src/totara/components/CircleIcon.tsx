/**
 * This file is part of Totara Enterprise.
 *
 * Copyright (C) 2020 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise is provided only to Totara Learning Solutions
 * LTDâ€™s customers and partners, pursuant to the terms and
 * conditions of a separate agreement with Totara Learning
 * Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [sales@totaralearning.com] for more information.
 */

import React from "react";
import { StyleSheet, View, Image } from "react-native";

import { iconSizes } from "@totara/theme/constants";
// @ts-ignore no types published yet for fortawesome react-native, they do have it react so check in future and remove this ignore
import { FontAwesomeIcon } from "@fortawesome/react-native-fontawesome";
import { TotaraTheme } from "@totara/theme/Theme";

type Props = {
  icon?: any;
  backgroundColor?: string;
  iconColor?: string;
  borderColor?: string;
  fontAwesomeIcon?: boolean;
};

const CircleIcon = ({ icon, backgroundColor, iconColor, borderColor, fontAwesomeIcon = true }: Props) => {
  return (
    <View
      style={[
        styles.iconCircle,
        {
          backgroundColor: backgroundColor == null ? TotaraTheme.textColorDark : backgroundColor,
          borderColor: borderColor == null ? TotaraTheme.textColorDark : borderColor
        }
      ]}>
      {icon &&
        (fontAwesomeIcon ? (
          <FontAwesomeIcon
            icon={icon}
            size={iconSizes.sizeM / 2}
            color={iconColor == null ? TotaraTheme.colorAccent : iconColor}
          />
        ) : (
          <Image source={icon}></Image>
        ))}
    </View>
  );
};

const styles = StyleSheet.create({
  iconCircle: {
    borderRadius: iconSizes.sizeM / 2,
    borderWidth: 1,
    height: iconSizes.sizeM,
    width: iconSizes.sizeM,
    borderStyle: "solid",
    justifyContent: "center",
    alignItems: "center"
  }
});

export default CircleIcon;
