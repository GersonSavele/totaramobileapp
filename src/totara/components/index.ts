/**
 * This file is part of Totara Enterprise.
 *
 * Copyright (C) 2019 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise is provided only to Totara Learning Solutions
 * LTDâ€™s customers and partners, pursuant to the terms and
 * conditions of a separate agreement with Totara Learning
 * Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [sales@totaralearning.com] for more information.
 */

import DueDateState from "../features/currentLearning/components/DueDateState";
import CircleIcon from "./CircleIcon";
import { AddBadge, CheckBadge } from "./AddBadge";
import ProgressCircle from "./ProgressCircle";
import InfoModal from "./InfoModal";
import AppStateListener from "./AppStateListener";
import PrimaryButton from "./PrimaryButton";
import SecondaryButton from "./SecondaryButton";
import GeneralErrorModal from "./GeneralErrorModal";
import InputTextWithInfo from "./InputTextWithInfo";
import TertiaryButton from "./TertiaryButton";
import TouchableIcon from "./TouchableIcon";
import FormError from "./FormError";
import MoreText from "./MoreText";
import LinkText from "./LinkText";
import MessageBar from "./MessageBar";
import TotaraNavigationOptions from "./NavigationOptions";
import LoadingError from "./LoadingError";
import NetworkStatusIndicator from "./NetworkStatusIndicator";
import Loading from "./Loading";
import ImageWrapper from "./ImageWrapper";
import ModalContent from "./ModalContent";
import WebviewToolbar from "./WebviewToolbar";

export {
  AddBadge,
  DueDateState,
  CircleIcon,
  CheckBadge,
  ProgressCircle,
  AppStateListener,
  PrimaryButton,
  GeneralErrorModal,
  InfoModal,
  InputTextWithInfo,
  TertiaryButton,
  TouchableIcon,
  FormError,
  SecondaryButton,
  MoreText,
  LinkText,
  MessageBar,
  TotaraNavigationOptions,
  Loading,
  LoadingError,
  NetworkStatusIndicator,
  ImageWrapper,
  ModalContent,
  WebviewToolbar
};
