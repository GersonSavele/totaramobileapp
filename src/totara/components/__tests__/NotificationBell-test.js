/**
 * This file is part of Totara Enterprise.
 *
 * Copyright (C) 2020 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise is provided only to Totara Learning Solutions
 * LTDâ€™s customers and partners, pursuant to the terms and
 * conditions of a separate agreement with Totara Learning
 * Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [sales@totaralearning.com] for more information.
 */

import React from "react";
import renderer from "react-test-renderer";
import { NotificationBell } from "@totara/components";

describe("NotificationBell", () => {
  const tintColor = "red"; //COLOR DOES NOT MAKES ANY DIFFERENCE

  it("active with zero notifications", async () => {
    const active = true;
    const counting = 0;
    const notificationBell = renderer.create(
      <NotificationBell active={active} tintColor={tintColor} counting={counting} />
    );
    expect(notificationBell.toJSON()).toMatchSnapshot();
  });

  it("active with 10 notifications", async () => {
    const active = true;
    const counting = 10;
    const notificationBell = renderer.create(
      <NotificationBell active={active} tintColor={tintColor} counting={counting} />
    );
    expect(notificationBell.toJSON()).toMatchSnapshot();
  });

  it("inactive with 100 notifications", async () => {
    const active = false;
    const counting = 100;
    const notificationBell = renderer.create(
      <NotificationBell active={active} tintColor={tintColor} counting={counting} />
    );
    expect(notificationBell.toJSON()).toMatchSnapshot();
  });
});
