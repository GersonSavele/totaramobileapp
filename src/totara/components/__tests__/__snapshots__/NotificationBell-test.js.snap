// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationBell active with 10 notifications 1`] = `
<View
  style={
    Object {
      "width": 36,
    }
  }
>
  <View>
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/resources/images/tabbar/notificationsSolid.png",
        }
      }
      style={
        Array [
          Object {
            "height": 24,
            "width": 24,
          },
          Object {
            "tintColor": "red",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "justifyContent": "flex-end",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "red",
          "borderRadius": 8,
          "flex": 1,
          "justifyContent": "center",
          "paddingLeft": 6,
          "paddingRight": 6,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "white",
            "fontSize": 11,
          }
        }
      >
        10
      </Text>
    </View>
  </View>
</View>
`;

exports[`NotificationBell active with zero notifications 1`] = `
<View
  style={
    Object {
      "width": 36,
    }
  }
>
  <View>
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/resources/images/tabbar/notificationsSolid.png",
        }
      }
      style={
        Array [
          Object {
            "height": 24,
            "width": 24,
          },
          Object {
            "tintColor": "red",
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`NotificationBell inactive with 100 notifications 1`] = `
<View
  style={
    Object {
      "width": 36,
    }
  }
>
  <View>
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/resources/images/tabbar/notificationsRegular.png",
        }
      }
      style={
        Array [
          Object {
            "height": 24,
            "width": 24,
          },
          Object {
            "tintColor": "red",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "justifyContent": "flex-end",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "red",
          "borderRadius": 8,
          "flex": 1,
          "justifyContent": "center",
          "paddingLeft": 6,
          "paddingRight": 6,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "white",
            "fontSize": 11,
          }
        }
      >
        99+
      </Text>
    </View>
  </View>
</View>
`;
